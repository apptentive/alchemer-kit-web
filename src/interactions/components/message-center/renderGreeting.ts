/**
 * Renders the Message Center greeting (icon, title, body, info link).
 *
 * @param {string} titleText - The text to use as the section title
 * @param {string} bodyText - The text to use for the section content
 * @returns {HTMLDivElement} - The greeting DOM Node
 */
const renderGreeting = (titleText: string, bodyText: string) => {
  // NOTE: Currently unused.
  // const icon = document.createElement('img');
  // icon.className = 'apptentive-message-center-greeting-icon';
  // icon.src = this.sdk.i18n.translate(`${this.interaction.id}.configuration.greeting.image_url`);
  // icon.alt = '';
  // icon.width = 80;
  // icon.height = 80;
  // greeting.appendChild(icon);

  const title = document.createElement('h2');
  title.classList.add('apptentive-message-center-greeting__title');
  title.classList.add('apptentive-typography__header');
  title.textContent = titleText;

  const body = document.createElement('p');
  body.classList.add('apptentive-typography__body');
  body.textContent = bodyText;

  const greeting = document.createElement('div');
  greeting.classList.add('apptentive-message-center-greeting');
  greeting.append(title);
  greeting.append(body);

  return greeting;
};

export default renderGreeting;
