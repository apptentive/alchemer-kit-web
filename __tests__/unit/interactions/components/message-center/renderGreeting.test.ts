import renderGreeting from '../../../../../src/interactions/components/message-center/renderGreeting';

const _titleText = 'Title';
const _bodyText = 'Body content';

describe('renderGreeting', () => {
  test('properly renders blank', () => {
    const output = renderGreeting(_titleText, _bodyText);

    expect(output).toMatchSnapshot();
  });
});
