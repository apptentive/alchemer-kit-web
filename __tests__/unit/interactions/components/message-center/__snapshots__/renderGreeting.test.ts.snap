// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderGreeting properly renders blank 1`] = `
<div
  class="apptentive-message-center-greeting"
>
  <h2
    class="apptentive-message-center-greeting__title apptentive-typography__header"
  >
    Title
  </h2>
  <p
    class="apptentive-typography__body"
  >
    Body content
  </p>
</div>
`;
